aten/src/ATen/native/cuda/SparseFactories.cu

```diff
@@ -69,8 +69,36 @@ void _spdiags_kernel_cuda(
             });
       });
 }
+
+void _spdiags_backward_kernel_cuda(TensorIterator& iter, Tensor& grad_in) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND3(
+      at::ScalarType::BFloat16,
+      at::ScalarType::Half,
+      at::ScalarType::ComplexHalf,
+      iter.dtype(),
+      "spdiags_backward_cpu",
+      [&] {
+        auto* grad_in_ptr = grad_in.data_ptr<scalar_t>();
+        int64_t const grad_in_row_stride = grad_in.stride(0);
+        int64_t const grad_in_col_stride = grad_in.stride(1);
+        gpu_kernel(
+            iter,
+            [grad_in_ptr, grad_in_row_stride, grad_in_col_stride] GPU_LAMBDA(
```

Comment:
nit: we can reuse this kernel both for the CPU and CUDA. This, however, will require defining a `LAMBDA` that resolves to `GPU_LAMBDA`. You could do that under `#if defined(__CUDACC__) || defined(__HIPCC__)` if you want to go that route :)